### ISSUE TITLE
Upgrading CISCO SDK from 2.8 to 3.7

### ISSUE THREAD

**Issue by CDL24** [webex/webex-android-sdk#129]
Facing a problem with SpaceID. Does not getting spaceId most of the time after authorising JWT token. I have tested apk of kitchenSink app 3.7.0 in that also we are not getting spaceId many times. I have tested KitchenSink app 3.6.0 apk in that we are able to get spaceId.

**Comment by ralagana** [comment]
Can you provide us some details on exactly how you're trying this? Do you have code snippets to support this?

Also, have you tried reproducing this issue with our demo KitchenSink App https://github.com/webex/webex-android-sdk-example? If not, can you try it and let us know your results?

**Comment by CDL24** [comment]
Yes I have tried with KitchenSink 3.7.0 sample app and in that when we loggedIn successfully and go to 'Initiate Call' option and click on 'Space' Tab at that time getSpaces() method is called and return success but with 0 data. i have checked this multiple times but not getting spaceId. You can reproduce this issue in KitchenSink 3.7.0 apk (Already there in webex git page).

The same steps are working fine in KitchenSink 3.6.0 apk. In this getSpace() api will take some time to return spaceId but it will return spaceId.

**Comment by ralagana** [comment]
@CDL24 thanks for this information. We'll test this out and let you know our results.

**Comment by ralagana** [comment]
Hi @CDL24, sorry for the delay. We tried reproducing this but to be honest it's not yet clear exactly which class you're debugging this through. Can you provide us the class in KitchenSink App where you're debugging this method? Can you also share a screenshot of what you're seeing? Reason we ask is that there is no "getSpaces()" method in our KitchenSink app. We have getSpacesList() , getspaceById(), getSpace(), but not the one you mentioned. Let us know on these points please.

**Comment by CDL24** [comment]
Hi @ralagana below is method i am checking in KithenSink 3.7.0 app
*
fun fetchSpacesList(teamId: String?, maxSpaces: Int, sortBy: SortBy): Observable<List<SpaceModel>> {
        return Single.create<List<SpaceModel>> { emitter ->
            webex.spaces.list(teamId, maxSpaces, null, sortBy, CompletionHandler { result ->
                if (result.isSuccessful) {
                    emitter.onSuccess(result.data?.map {
                        SpaceModel.convertToSpaceModel(it)
                    } ?: emptyList())
                } else {
                    emitter.onError(Throwable(result.error?.errorMessage))
                }
            })
        }.toObservable()
    }
*

It will not return space. Even you can check in KitchenSink sample apk 3.7.0. After login into app just go to Spaces Tab there you will not get space.

**Comment by ralagana** [comment]
@CDL24 thanks for this info. I tested with the same KS app on v3.7 and debugged that method to find that it does in fact return the space ID for each space in the list. See here: https://www.evernote.com/shard/s375/sh/5ebf260d-2440-b706-3a94-7b6bdaf15fe7/f63d981d115307d579175b038cd55c04 for one example.

Note that the space ID is returned simply as "id" by this method, given that it's implied already with the method's name that this is for spaces. Are you not seeing this on your side? Please provide a screenshot of the same.
